system/stories/src/Checkbox.stories.tsx

```diff
@@ -21,15 +24,22 @@ export default {
 
 const Template: Story = ({ Checkbox, LabelComponent }) => (
   <>
-    {[true, false].map((isChecked) =>
+    {states.map((state) =>
       contentVariants.map((variant) => (
         <LabelComponent>
           <Checkbox
             data-pseudo={variant.toLowerCase()}
             disabled={variant.toLowerCase() === 'disabled'}
-            defaultChecked={isChecked}
+            ref={(ref: undefined | HTMLInputElement) => {
+              if (!ref) return;
+              if (state === indeterminate) {
+                ref.indeterminate = true;
+              } else {
+                ref.checked = state;
+              }
+            }}
           />
-          {variant} {isChecked ? '☑' : '☐'}
+          {variant} {state ? '☑' : '☐'}
```

Comment:
Just one small tweak (and yes, usually nested ternaries is bad, but here... meh. If you feel like it extract to a function)
```suggestion
          {variant} {state === 'true' ? '☑' : state === false ? '☐' : '-'}
```